corsheaders/defaults.py

```diff
@@ -21,28 +21,50 @@
     'DELETE',
     'OPTIONS',
 )
-CORS_ALLOW_METHODS = getattr(settings, 'CORS_ALLOW_METHODS', default_methods)
-
-CORS_ALLOW_CREDENTIALS = getattr(settings, 'CORS_ALLOW_CREDENTIALS', False)
-
-CORS_PREFLIGHT_MAX_AGE = getattr(settings, 'CORS_PREFLIGHT_MAX_AGE', 86400)
-
-CORS_ORIGIN_ALLOW_ALL = getattr(settings, 'CORS_ORIGIN_ALLOW_ALL', False)
-
-CORS_ORIGIN_WHITELIST = getattr(settings, 'CORS_ORIGIN_WHITELIST', ())
 
-CORS_ORIGIN_REGEX_WHITELIST = getattr(
-    settings,
-    'CORS_ORIGIN_REGEX_WHITELIST',
-    ())
 
-CORS_EXPOSE_HEADERS = getattr(settings, 'CORS_EXPOSE_HEADERS', ())
+class CorsHeadersSettings(object):
+    CORS_ALLOW_HEADERS = default_headers
+    CORS_ALLOW_METHODS = default_methods
+    CORS_ALLOW_CREDENTIALS = False
+    CORS_PREFLIGHT_MAX_AGE = 86400
+    CORS_ORIGIN_ALLOW_ALL = False
+    CORS_ORIGIN_WHITELIST = ()
+    CORS_ORIGIN_REGEX_WHITELIST = ()
+    CORS_EXPOSE_HEADERS = ()
+    CORS_URLS_REGEX = '^.*$'
+    CORS_MODEL = None
+    CORS_REPLACE_HTTPS_REFERER = False
 
-CORS_URLS_REGEX = getattr(settings, 'CORS_URLS_REGEX', '^.*$')
+    def __init__(self, config, override=None):
+        if override is None:
+            override = {}
+        fields = [
+            'CORS_ALLOW_HEADERS',
+            'CORS_ALLOW_METHODS',
+            'CORS_ALLOW_CREDENTIALS',
+            'CORS_PREFLIGHT_MAX_AGE',
+            'CORS_ORIGIN_ALLOW_ALL',
+            'CORS_ORIGIN_WHITELIST',
+            'CORS_ORIGIN_REGEX_WHITELIST',
+            'CORS_EXPOSE_HEADERS',
+            'CORS_URLS_REGEX',
+            'CORS_MODEL',
+            'CORS_REPLACE_HTTPS_REFERER',
+        ]
```

Comment:
you could maybe get that list with

```python
fields = [field for field in dir(CorsHeadersSettings) if field.startswith('CORS_')]
```

to avoid adding a new config value but not adding it in that fields list.